### DISCUSSION TITLE
Products and payments in Casdoor

### DISCUSSION THREAD

**Discussion by ISulimanov** [casdoor/casdoor#1674] [Q&A] [answered]
It is very surprising to find the presence of such functionality as **products** and **payment providers**, since they are not directly related to IAM

can you tell a little why you decided to include it in the main part of the product and what plans you have to develop this functionality?

**Comment by hsluoyz** [comment, marked as answer]
@ISulimanov yes, those are not very classic IAM functionality. But we add it because some people may need their Casdoor users to do payments. In this scenario, Casdoor acts as a "gateway" for payments, just like Email, SMS, OAuth, etc. Actually Casdoor "can" be used merely as a Email or SMS forwarder without using IAM (OAuth, OIDC, etc.) at all. It's a broader concept than a normal IAM. Fortunately it fits into our provider model well and doesn't bring too much trouble. So we kept that functionality.

If you don't need it, you can simply disable it at:

https://github.com/casdoor/casdoor/blob/14fa914e6f7f9fe045ebb53606e81853f3d8e73b/web/src/Conf.js#L22
